### ISSUE TITLE
[B16] Rotten Flesh not having aspects in modpacks

### ISSUE THREAD

**Issue by Plet53** [Azanor/thaumcraft-beta#975]
Rotten Flesh does not have any aspects, nor does its block form, with the following modset installed:
Baubles [1.5.2]
EnderCore [0.5.30]
EnderIO [5.0.26]
Mantle [1.3.2.24]
Natura [4.3.2.49]
Thaumcraft [6.1.BETA16]
Forge [14.23.4.2707]

**Comment by Plet53** [comment]
SleepyTrousers/EnderIO#4688
progwml6/Natura#452

**Comment by tyler489** [comment]
Neither does soulsand or snow or anythjng made from these items.. oddly enough they appear in jei as having the aspects

**Comment by Jamie787** [comment]
I also have the same bug i'm pretty sure it's to do with enderio because when i got rid of it the aspects came back

for me coal is missing as well

**Comment by Elesrath** [comment]
TLDR: 
EnderCore [0.5.31] + Natura [4.3.2.49] = No aspects on Rotten Flesh and Books
EnderCore [0.5.31] + EnderIO [5.0.26] = No aspects on Coal, Coal Blocks and Redstone Blocks.

It happens for me with only (Forge 14.23.4.2708 - Newest version as of posting) :
Baubles [1.5.2]
EnderCore [0.5.30 or 0.5.31]
EnderIO [5.0.26]
Thaumcraft [6.1.BETA16]

The items that are missing aspects that I have encountered are:
Coal
Coal Blocks
Redstone Blocks

However, I do have aspects on rotten flesh.

Removing just Ender IO brings the aspects back, even in an existing save.

After some more testing, I have found the following:

Using these setups no items lose their aspects:
Baubles [1.5.2]
Thaumcraft [6.1.BETA16]
Mantle [1.3.2.24]
Natura [4.3.2.49]

Baubles [1.5.2]
Thaumcraft [6.1.BETA16]
EnderCore [0.5.31]

Baubles [1.5.2]
Thaumcraft [6.1.BETA16]
Mantle [1.3.2.24]

Baubles [1.5.2]
Thaumcraft [6.1.BETA16]
EnderCore [0.5.31]
Mantle [1.3.2.24]

However, with this setup Rotting Flesh and Books lose their aspects, but Coal, Coal Blocks and Redstone Blocks don't:
EnderCore [0.5.31]
Baubles [1.5.2]
Thaumcraft [6.1.BETA16]
Mantle [1.3.2.24]
Natura [4.3.2.49]

Using this setup all of the items I've mentioned lose their aspects:
Baubles [1.5.2]
EnderCore [0.5.31]
EnderIO [5.0.26]
Thaumcraft [6.1.BETA16]
Mantle [1.3.2.24]
Natura [4.3.2.49]

So my conclusion would be that EnderCore [0.5.31] + Natura [4.3.2.49] = No aspects on Rotten Flesh and Books, and using EnderCore [0.5.31] + EnderIO [5.0.26] means no aspects on Coal, Coal Blocks and Redstone Blocks.

**Comment by Elesrath** [comment]
After some further testing, I've discovered that the issue was introduced with EnderIO 5.0.25 & EnderCore 0.5.30. The previous version down, EnderIO 5.0.24 & EnderCore 0.5.22, does not have this issue. Further, Running EnderIO 5.0.24 & EnderCore 0.5.30 also does not cause the issue, so I conclude that the issue is being caused by something introduced in EnderIO 5.0.25

**Comment by Elesrath** [comment]
Actually, looking at the changelog for EnderIO 5.0.25, I'm going to assume that this is the issue:

> Added Thaumcraft integration. This includes aspects, infusion stabilising endermen heads and darksteel upgrades.

**Comment by tyler489** [comment]
Interesting.. i asked about that a few days ago and they said they tested with that off.

**Comment by jdratlif** [comment]
After downgrading to EnderIO 5.0.24, I have aspects on coal again.

**Comment by xenanthropy** [comment]
I downgraded to EnderIO 5.0.24 and EnderCore 0.5.22 and i'm still missing aspects on coal, snow, and rotten flesh. Currently running Natura 4.3.2.49

**Comment by Elesrath** [comment]
This is still occurring with the newest versions:
EnderCore-1.12.2-0.5.35
EnderIO-1.12.2-5.0.29
Thaumcraft-1.12.2-6.1.BETA17

**Comment by tyler489** [comment]
@azanor evidently the coal losing aspect comes from enderio doing

 ThaumcraftApi.registerObjectTag("dustCoal", new AspectList().add(getAspects(Items.COAL))); 
I changed .add(getAspects(Items.COAL)) to .add(Aspect.FIRE, 10).add(Aspect.ENERGY, 10) and the normal coal got its aspects back, is getAspects buggy somehow when used like this?

**Comment by Lilylicious** [comment]
Having Thaumic Equivalency and EvilCraft in a pack causes coal to lose aspects. Removing EvilCraft after this happens makes them regain it, but removing Thaumic Equivalency has no effect.

**Comment by tterrag1098** [comment]
Pardon my ignorance, but is there a better way?
